scripts/demo/sv4d_helpers.py

```diff
@@ -158,46 +154,120 @@ def read_video(
-                image.thumbnail([W, H], Image.Resampling.LANCZOS)
+            if image.mode == "RGBA":
+                pass
+            else:
+                # image.thumbnail([W, H], Image.Resampling.LANCZOS)
                 image = remove(image.convert("RGBA"), alpha_matting=True)
-            image_arr = np.array(image)
-            in_w, in_h = image_arr.shape[:2]
+            images[i] = image
+
+    # Crop video frames, assume the object is already in the center of the image
+    white_thresh = 250
+    images_v0 = []
+    box_coord = [np.inf, np.inf, 0, 0]
+    for image in images:
+        image_arr = np.array(image)
+        in_w, in_h = image_arr.shape[:2]
+        original_center = (in_w // 2, in_h // 2)
+        if image.mode == "RGBA":
             ret, mask = cv2.threshold(
                 np.array(image.split()[-1]), 0, 255, cv2.THRESH_BINARY
             )
-            x, y, w, h = cv2.boundingRect(mask)
-            max_size = max(w, h)
-            if t == 0:
-                side_len = (
-                    int(max_size / image_frame_ratio)
-                    if image_frame_ratio is not None
-                    else in_w
-                )
-            padded_image = np.zeros((side_len, side_len, 4), dtype=np.uint8)
-            center = side_len // 2
-            padded_image[
-                center - h // 2 : center - h // 2 + h,
-                center - w // 2 : center - w // 2 + w,
-            ] = image_arr[y : y + h, x : x + w]
-            rgba = Image.fromarray(padded_image).resize((W, H), Image.LANCZOS)
-            rgba_arr = np.array(rgba) / 255.0
-            rgb = rgba_arr[..., :3] * rgba_arr[..., -1:] + (1 - rgba_arr[..., -1:])
-            image = Image.fromarray((rgb * 255).astype(np.uint8))
         else:
-            image = image.convert("RGB").resize((W, H), Image.LANCZOS)
-        image = ToTensor()(image).unsqueeze(0).to(device)
-        images_v0.append(image * 2.0 - 1.0)
-    return images_v0
-
+            # assume the input image has white background
+            ret, mask = cv2.threshold(
+                (np.array(image).mean(-1) <= white_thresh).astype(np.uint8) * 255, 0, 255, cv2.THRESH_BINARY
+            )
+            
+        x, y, w, h = cv2.boundingRect(mask)
+        box_coord[0] = min(box_coord[0], x)
+        box_coord[1] = min(box_coord[1], y)
+        box_coord[2] = max(box_coord[2], x + w)
+        box_coord[3] = max(box_coord[3], y + h)
+    box_square = max(original_center[0] - box_coord[0], original_center[1] - box_coord[1])
+    box_square = max(box_square, box_coord[2] - original_center[0])
+    box_square = max(box_square, box_coord[3] - original_center[1])
+    x, y, w, h = original_center[0] - box_square, original_center[1] - box_square, 2 * box_square, 2 * box_square
+    box_size = box_square * 2
+
+    for image in images:
+        if image.mode == "RGB":
+            image = image.convert("RGBA")
+        image_arr = np.array(image)
+        side_len = (
```

Comment:
Is this image padding temporally consistent? Please check the latest commit where we calculate the "side_len" (bbox width and height) in the first frame and use it for all the remaining frames.

Reply:
Yes it's temporally consistent. We calculate a fixed box that can cover object in all frames.